### PR TITLE
update multi-tenancy design doc about forced-sharing CRD

### PR DESCRIPTION
**Pull request by chenqianfzh**
Update the doc about forced-sharing system CRD.

### CODE REVIEW

File: docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md [hunk 1]
```diff
@@ -168,14 +169,22 @@ Here are some examples:
...
   * Additionally, the applicable range of a CRD can be changed by the cluster operator at any time in a simple and quick manner. That will be useful to CRD test and rolling update.  
 
-  (*** What the managebility should look like is not finalized as we are still working on it. )
+
+  
+  When a system CRD is usable to regular tenants, this sharing can be:
+  * **forced**. The system CRD will be usable by all the tenants. It overrides the local CRDs with the same name in non-system tenant spaces. Once a forced system CRD is deployed, a regular tenant's attempt to create a CRD with the same name will be rejected. 
+
+    Cluster admins are suggested to use it only for those resource types mandatory to every tenant.  
+  * **optional**. The regular tenant is allowed to override such system CRDs by defining a local CRD.
```
Comment (h-w-chen) [inline docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md:181]: so this "optional"  sounds like "tenant first"? If that is supported, what would be the results of get such CR --tenant=all?

File: docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md [hunk 2]
```diff
@@ -168,14 +169,22 @@ Here are some examples:
...
  
  Note: it does not contradict the principle of isolation. The isolation principle is about two regular tenants, while this principle is about the system tenant and regular tenants. 
 
- 3. **Autonomy**. It means that the tenant can install his own CRD and deploy the resources and CRD operators within his own tenant's world, without the help from the cluster admins. Autonomy also means that the tenant admin has the final say on what CRD will be used in his tenant space when there is an overlap between the tenant's CRD and that of the system tenant. The tenant can choose MultiTenancyCRD policy to be :
+ 3. **Autonomy**. It means that the tenant can install his own CRD and deploy the resources and CRD operators within his own tenant's space, without the help from the cluster admins. Autonomy also means that the tenant admin has has control on what CRD will be used in his space when there is an collision between the tenant's CRD and that of the system tenant (surely, the system CRD is not forced). The tenant can choose MultiTenancyCRD policy to be :
```
Comment (h-w-chen) [inline docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md:187]: can this "autonomy" be overruled by "forced"?

File: docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md [hunk 3]
```diff
@@ -26,14 +26,14 @@ Below are some examples resources in a space for tenant "t1":
...
-As we can see, spaces provides an extra layer of resource separation. Under a certain space, a tenant can freely create their own resources like namespaces or PVs. They don't need to worry about naming conflicts.
+As we can see, spaces provide an extra layer of resource separation. Within his space, a tenant can freely create his own resources like namespaces or PVs. They don't need to worry about naming conflicts.
 
 If an API resource is not under any tenant space, then it's in the **system space**. There is only one system space in the whole cluster, which by default is only accessible to system tenant users.
 
 Example resources in the system space:
 
->- **/api/v1/namespaces/default/pods/pod1**
->- **/api/v1/persistentvolumes/pv1**
+>- **/api/v1/tenants/system/amespaces/default/pods/pod1**
```
Comment (XiaoningDing) [inline docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md:35]: "namespaces"

File: docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md [hunk 4]
```diff
@@ -204,32 +214,40 @@ However, we need to point out that, certain resources are not accessible from re
...
+
+  **arktos.futurewei.com/crd-sharing-policy: forced**
+ 
 * the logic to find a matching CRD for a request in the API server will be changed as the following:
 
 ```
 GetMatchingCRD(G/V/K, tenant) (found_CRD, error)
+  if (found a system forced sharing CRD) {
+    return (the system CRD), nil
+  }
```
Comment (XiaoningDing) [inline docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md:227]: So "SystemForced" CRDs will be return even if the tenant sets a policy of "NeverUseSystemCRD"?

File: docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md (hunk 3 above)
Comment (chenqianfzh) [reply docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md:35]: Fixed.

File: docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md (hunk 4 above)
Comment (chenqianfzh) [reply docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md:227]: updated.

File: docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md (hunk 1 above)
Comment (chenqianfzh) [reply docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md:181]: When list CR --tennat=all, we should only list the CR from the tenants that uses this system CRD. The tenants, who chooses to use their own CRD, actually are using a different type of resource.

As mentioned in the doc, only "forced" CRD's behavior is implemented so far. So the behavior for those optional CRDs are subject to change per real use scenarios.

**Pull request merged**